Components/DataBroker/Inc/DataBroker.hpp

```diff
@@ -0,0 +1,173 @@
+/**
+ ********************************************************************************
+ * @file    DataBroker.hpp
+ * @author  shivam
+ * @date    Nov 23, 2024
+ * @brief
+ ********************************************************************************
+ */
+
+#ifndef DATA_BROKER_HPP_
+#define DATA_BROKER_HPP_
+
+/************************************
+ * INCLUDES
+ ************************************/
+#include "Publisher.hpp"
+#include "SensorDataTypes.hpp"
+#include "Command.hpp"
+#include "DataBrokerMessageTypes.hpp"
+#include "SystemDefines.hpp"
+#include "Mutex.hpp"
+#include <type_traits>
+#include <cstring>
+
+/************************************
+ * MACROS AND DEFINES
+ ************************************/
+
+/************************************
+ * TYPEDEFS
+ ************************************/
+
+/************************************
+ * CLASS DEFINITIONS
+ ************************************/
+class DataBroker {
+ public:
+  /**
+   * @brief Publish data of a certain type
+   *         NOTE: You must ensure that there is a publisher for that type
+   */
+  template <typename T>
+  static void Publish(T* dataToPublish) {
```

Comment:
Consider making a static instance of the class instead of making everything in the class static